### PR TITLE
Use hash tree root for Node

### PR DESCRIPTION
**Pull request by ChihChengLiang**
This builds on top of #1077

### What was wrong?

Node should request and handle blocks with hash_tree_roots 

### How was it fixed?

- add chain.get_block_by_hash_tree_root
- fix node

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: eth2/beacon/db/chain.py [hunk 1]
```diff
@@ -612,6 +640,18 @@ def _add_block_root_to_slot_lookup(db: DatabaseAPI, block: BaseBeaconBlock) -> N
...
+    @staticmethod
+    def _add_block_hash_tree_root_to_signing_root_lookup(
+        db: DatabaseAPI, block: BaseBeaconBlock
+    ) -> None:
+        hash_tree_root_to_signing_root_key = SchemaV1.make_block_hash_tree_root_to_signing_root_lookup_key(
+            block.hash_tree_root
+        )
+        db.set(
+            hash_tree_root_to_signing_root_key,
+            ssz.encode(block.signing_root, sedes=ssz.sedes.bytes32),
```
Comment (NIC619) [inline eth2/beacon/db/chain.py]: I wonder do we need to ssz encode block root? (Or we simply ssz encode every value?)
If that's the case, after a quick lookup, we seem to be storing plain block root for keys `make_canonical_head_root_lookup_key`, `make_finalized_head_root_lookup_key` and `make_justified_head_root_lookup_key`. Do you mind doing a quick fix on that along in this PR?

Comment (ChihChengLiang) [reply eth2/beacon/db/chain.py]: We don't really have to `ssz.encode(block.signing_root, sedes=ssz.sedes.bytes32)` the result doesn't change. Should get rid of the unnecessary encodes.

File: eth2/beacon/db/chain.py [hunk 2]
```diff
@@ -93,6 +93,11 @@ def get_block_by_root(
     def get_slot_by_root(self, block_root: SigningRoot) -> Slot:
         pass
 
+    def get_block_signing_root_by_hash_tree_root(
```
Comment (mhchia) [inline eth2/beacon/db/chain.py]: Missing `abstractmethod`

File: .flake8-eth2 [hunk 3]
```diff
@@ -1,4 +1,4 @@
 [flake8]
-max-line-length= 100
+max-line-length= 120
```
Comment (mhchia) [inline .flake8-eth2]: Is there specific reason to have this change?

File: eth2/beacon/db/chain.py (hunk 2 above)
Comment (ChihChengLiang) [reply eth2/beacon/db/chain.py]: Nice catch!

File: .flake8-eth2 (hunk 3 above)
Comment (ChihChengLiang) [reply .flake8-eth2]: The file still has lines exceed 100 characters after running black. I'll remove this change and do future investigation.

**Pull request merged**